src/Stratis.Bitcoin.Features.BlockStore/BlockRepository.cs

```diff
@@ -58,55 +68,81 @@ public interface IBlockRepository : IBlockStore
         /// <param name="txIndex">Whether to index transactions.</param>
         Task SetTxIndexAsync(bool txIndex);
 
+        /// <summary>Hash and height indicating where the node's block store has been pruned up to.</summary>
+        HashHeightPair PrunedTip { get; }
+
+        /// <summary>
+        /// Sets the pruned tip.
+        /// <para> 
+        /// It will be saved once the block database has been compacted on node initialization or shutdown.
+        /// </para>
+        /// </summary>
+        /// <param name="tip">The tip to set.</param>
+        void UpdatePrunedTip(ChainedHeader tip);
+
         /// <summary>Hash and height of the repository's tip.</summary>
         HashHeightPair TipHashAndHeight { get; }
 
+        /// <summary> Indicates that the node should store all transaction data in the database.</summary>
         bool TxIndex { get; }
+
+        /// <summary>
+        /// Compacts the block and transaction database by resaving the database file without
+        /// all the deleted references.
+        /// </summary>
+        /// <param name="consensusTip">The current tip of consensus.</param>
+        /// <param name="nodeInitializing">Indicates whether or not this method is called from node startup or not.</param>
+        Task PruneDatabase(ChainedHeader consensusTip, bool nodeInitializing);
     }
 
     public class BlockRepository : IBlockRepository
     {
         private const string BlockTableName = "Block";
 
-        private const string TransactionTableName = "Transaction";
-
         private const string CommonTableName = "Common";
 
-        /// <summary>Instance logger.</summary>
-        private readonly ILogger logger;
+        /// <summary> The amount of blocks to keep from the block store's tip.</summary>
+        private const int PruneBlockMargin = 1000;
```

Comment:
Probably should be configurable

Reply:
on bitcoin, it's 288 blocks.

Reply:
By default, yes. And this value came from amount of time to produce 288 blocks. For stratis it takes 10 times less time to produce 288 blocks so it will make sense to have value of 2880. 
Also we need to remember that we can't prune more than MaxReorg blocks.

Reply:
OK I will change to 2880 👍

Reply:
Also please make it configurable

Reply:
Put it in StoreSettings?

Reply:
Yeah, sounds good

Reply:
Some info form the release notes (might be out of date though)
https://github.com/bitcoin/bitcoin/blob/v0.11.0rc1/doc/release-notes.md#block-file-pruning

The pruned value was always planned to go in the settings, but we need to set a minimum of pruned blocks, it just seemed to me that max reorg is too much of a close call, but besides that a pruned node will not signal Network Services so will never be asked to serve blocks, bitcoin by minimum prunes anything after 2 days of blocks, and the value it seems is based on mb size not.

So either we add a minimum to the network class or we use a multiple of max reorg (which actually might be disabled)